pkg/proxy/handler.go

```diff
@@ -158,5 +131,37 @@ func newReverseProxy(host string) *httputil.ReverseProxy {
 			AdditionalProxyRewrite(r)
 		},
 	}
-	return proxy
+
+	proxy.ModifyResponse = func(r *http.Response) error {
+		if h.isRetryCode(r.StatusCode) {
+			// Returning an error will trigger the ErrorHandler.
+			return errors.New("retry")
+		}
+		return nil
+	}
+
+	proxy.ErrorHandler = func(w http.ResponseWriter, r *http.Request, err error) {
+		if err != nil && pr.attempt < h.MaxRetries {
+			log.Println("retrying")
+			pr.attempt++
+			h.proxyHTTP(w, pr)
+			return
+		}
+
+		pr.sendErrorResponse(w, http.StatusBadGateway, "proxy: exceeded retries: %v/%v", pr.attempt, h.MaxRetries)
```

Comment:
IMHO the error message contains details that should not be shared. The retry mechanism is vulnerable for DoS attacks.

Reply:
This error message will not reach the client because the status code is `> 500`... See the convo here: https://github.com/substratusai/lingo/pull/61/files/a9bc8b12c6b49198b1799bf0ecc0763ab60c680a#r1464478674 ... Perhaps this is a little unexpected and should be refactored. Agreed, retry mechanism can create more load than needed. I think DoS attacks can be mitigated at ingress layers before they reach Lingo. Thoughts?